pandas/core/dtypes/cast.py

```diff
@@ -1085,3 +1085,24 @@ def cast_scalar_to_array(shape, value, dtype=None):
     values.fill(fill_value)
 
     return values
+
+
+def _maybe_convert_indexer(indexer, until):
+    """
+    Convert slice, tuple, list or scalar "indexer" to 1-d array of indices,
```

Comment:
I think it would be ok to first do a PR to add a level to our indexing structure of code, IOW

``pandas.core.indexing`` -> ``pandas.core.indexing.indexing``, then you add other help modules as needed
e.g. ``pands.core.indexing.cast`` for example (where things like this could live).